controllers/account.js

```diff
@@ -102,16 +112,22 @@ module.exports.updateProfile = async (req, res) => {
       filename: req.file.filename,
     };
   }
+
   await User.findByIdAndUpdate(req.user._id, profileData);
   res.redirect("/account");
 };
 
 module.exports.viewProfile = async (req, res) => {
   const { id } = req.params;
   const profileData = await fetchProfileData(id);
+  const jsonData = encodeURIComponent(JSON.stringify(profileData));
+  const highscores = encodeURIComponent(JSON.stringify(profileData.bestScores));
```

Comment:
I'm not sure why `highscores` is declared independently from `jsonData`, can you comment on this?

Reply:
I am not 100% sure why this is the case, but when I try to access the `bestScores` object that gets passed together with the rest of `profileData`, the object lacks keys that have no value, which is not something that I want. The passed `highscores` object, on the other hand, keeps those empty keys. 

This is how it looks if I try to console it out (highscores first, `profileData.bestScores` under it):
![image](https://github.com/DmitrijSidorkin/pixel-thing/assets/100046819/8611d38b-0a9b-4d81-8d24-9f67d5749d62)

Reply:
This affects the rendering of the highscores table in the account page, therefore it is important that the empty keys stay.